src/vs/workbench/contrib/extensions/browser/extensionEnablementWorkspaceTrustTransitionParticipant.ts

```diff
@@ -38,9 +39,11 @@ export class ExtensionEnablementWorkspaceTrustTransitionParticipant extends Disp
 							if (environmentService.remoteAuthority) {
 								hostService.reload();
 							} else {
-								extensionService.stopExtensionHosts(true); // TODO@lszomoru adopt support for extension host to veto stopping
+								const stopped = await extensionService.stopExtensionHosts(localize('restartExtensionHost.reason', "Restarting extension host due to workspace trust change."));
 								await extensionEnablementService.updateExtensionsEnablementsWhenWorkspaceTrustChanges();
```

Comment:
@lszomoru here I was assuming that I can still let the `updateExtensionsEnablementsWhenWorkspaceTrustChanges` call run, even when the ext host was not stopped. Since in the above case for `untrusted` => `trusted` we call this method without restart.

Reply:
@lszomoru got it. a partial "fix" would be to still `await` the result but not support stopping it. In 99% of the cases I would assume that there will be no veto because in custom editors and notebooks we simply save the editors and only return `false` if that fails.

Reply:
@bpasero, that sounds reasonable to me until we add veto support for workspace trust transition.